xen/drivers/passthrough/iommu.c

```diff
@@ -129,7 +129,7 @@ static void __init parse_iommu_param(char *s)
     } while ( ss );
 }
 
-int iommu_domain_init(struct domain *d)
+int iommu_domain_init(struct domain *d, bool_t use_iommu)
```

Comment:
Next the patches were squashed.
1.1 "iommu: Call iommu construct() if use_iommu flag is set"
1.2 "iommu: Pass additional use_iommu argument to iommu_domain_init()".

Following text was added to patch description:
    Basically, this patch is needed for unshared IOMMUs on ARM only
    since the unshared IOMMUs on x86 are ok if iommu_construct() is called
    later. But, in order to be more generic and for possible future optimization
    make this change appliable for both platforms.
    So, the patch target is to make ARM happy and not to brake x86.
    Confirmation from x86 guys is needed.